---
title: Estimating protein hydrodynamic parameters and solution properties
excerpt: Solution properties of proteins from primary sequence and structural coordinates.
classes: wide
output:
  md_document:
    variant: gfm
    preserve_yaml: true
knit: (function(input, ...) {
    rmarkdown::render(
      input,
      output_file = paste0(
        '/Users/gbedwell/Documents/github/gbedwell.github.io/_posts/', '2023-02-10', '-', 'hydroprot','.md'
      ),
      envir = globalenv()
    )
  })
---

```{r, include=FALSE}
date <- "2023-02-10"

base.dir <- "/Users/gbedwell/Documents/github/gbedwell.github.io/"
base.url <- "/"
dir.create(paste0("../figures/", date))
fig.path <- paste0("../figures/", date)

knitr::opts_chunk$set(
  fig.asp = 0.618,
  fig.width = 6,
  dpi = 300,
  fig.align = "center",
  out.width = "80%",
  fig.cap = "center",
  comment = "#>",
  collapse = TRUE,
  dev = "png",
  base.dir = base.dir, 
  base.url = base.url,
  fig.path = fig.path
  )
```

To use <code>gget</code> in R, you'll need to install both <code>gget</code> and the R package <code>reticulate</code> (unless you're using [quarto](https://quarto.org/), but I am not in this post). <code>gget</code> is described in more detail [here](https://github.com/pachterlab/gget). If you use <code>gget</code>, please cite the appropriate [manuscript](https://academic.oup.com/bioinformatics/article/39/1/btac836/6971843).

```{r, eval=FALSE}
system("pip install gget")
install.packages("reticulate")
```

```{r}
library(reticulate)
use_condaenv(condaenv = 'python310', required = TRUE)
gget <- import("gget")
```

```{r}
gene.info <- gget$search("CPSF6", "homo_sapiens")
gene.info
```

```{r}
target.ensembl <- gene.info$ensembl_id[1]
uniprot.id <- gget$info(target.ensembl)$uniprot_id
seq <- gget$seq(target.ensembl, translate=TRUE)[2]
```

```{r}
lengths(regmatches(c("ADRNP"), gregexpr("A", c("ADRNPA"))))
```

```{r}
library(BSgenome.Hsapiens.NCBI.T2T.CHM13v2.0)

genome.obj <- BSgenome.Hsapiens.NCBI.T2T.CHM13v2.0
target.chr <- "19"
chr.length <- GenomeInfoDb::seqlengths(genome.obj)[[target.chr]]
n.sites <- 4e6
chr.vec <- sample(target.chr, n.sites, replace = TRUE)
site.pos <- sample(x = 1:chr.length, size = n.sites, replace = TRUE)
site.strand <- sample(x=c("+","-"), size=n.sites, replace=TRUE)
sites <- GenomicRanges::GRanges(seqnames=chr.vec,
                                ranges=IRanges::IRanges(start=site.pos, width=1),
                                strand=site.strand)
```



















