<!doctype html>
<!--
  Minimal Mistakes Jekyll Theme 4.24.0 by Michael Rose
  Copyright 2013-2020 Michael Rose - mademistakes.com | @mmistakes
  Free for personal and commercial use under the MIT license
  https://github.com/mmistakes/minimal-mistakes/blob/master/LICENSE
-->
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin _includes/seo.html --><title>Curriculum vitae - Signal From Noise</title>
<meta name="description" content="">


  <meta name="author" content="Greg Bedwell">
  


<meta property="og:type" content="website">
<meta property="og:locale" content="en_US">
<meta property="og:site_name" content="Signal From Noise">
<meta property="og:title" content="Curriculum vitae">
<meta property="og:url" content="http://localhost:4000/cv.html">


  <meta property="og:description" content="">











  

  


<link rel="canonical" href="http://localhost:4000/cv.html">




<script type="application/ld+json">
  {
    "@context": "https://schema.org",
    
      "@type": "Person",
      "name": "Greg Bedwell",
      "url": "http://localhost:4000/"
    
  }
</script>







<!-- end _includes/seo.html -->



  <link href="/feed.xml" type="application/atom+xml" rel="alternate" title="Signal From Noise Feed">


<!-- https://t.co/dKP3o1e -->
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="/assets/css/main.css">
<link rel="preload" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5/css/all.min.css" as="style" onload="this.onload=null;this.rel='stylesheet'">
<noscript><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5/css/all.min.css"></noscript>



    <!-- start custom head snippets -->

<!-- insert favicons. use https://realfavicongenerator.net/ -->

<!-- end custom head snippets -->

  </head>

  <body class="layout--single">
    <nav class="skip-links">
  <ul>
    <li><a href="#site-nav" class="screen-reader-shortcut">Skip to primary navigation</a></li>
    <li><a href="#main" class="screen-reader-shortcut">Skip to content</a></li>
    <li><a href="#footer" class="screen-reader-shortcut">Skip to footer</a></li>
  </ul>
</nav>

    

<div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
        
        <a class="site-title" href="/">
          Signal From Noise
          
        </a>
        <ul class="visible-links"><li class="masthead__menu-item">
              <a href="/year-archive/">Posts</a>
            </li><li class="masthead__menu-item">
              <a href="/about.html">About Me</a>
            </li><li class="masthead__menu-item">
              <a href="/cv.html">CV</a>
            </li></ul>
        
        <button class="greedy-nav__toggle hidden" type="button">
          <span class="visually-hidden">Toggle menu</span>
          <div class="navicon"></div>
        </button>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>


    <div class="initial-content">
      





<div id="main" role="main">
  


  <article class="page h-entry" itemscope itemtype="https://schema.org/CreativeWork">
    <meta itemprop="headline" content="Curriculum vitae">
    <meta itemprop="description" content="">
    
    

    <div class="page__inner-wrap">
      
        <header>
          <h1 id="page-title" class="page__title p-name" itemprop="headline">
            <a href="http://localhost:4000/cv.html" class="u-url" itemprop="url">Curriculum vitae
</a>
          </h1>
          


        </header>
      

      <section class="page__content e-content" itemprop="text">
        
        <h2 id="-education"><i class="fas fa-graduation-cap"></i> Education</h2>

<ul>
  <li>2021 - Post-doctoral fellow, Dana-Farber Cancer Institute/Harvard Medical School.</li>
  <li>2016 - Ph.D., Biophysics, University of Alabama at Birmingham.
    <ul>
      <li>Dissertation: Exploring bacteriophage P22 as a selective molecular scaffold and molecular sensor.</li>
    </ul>
  </li>
  <li>2010 - B.S., Chemistry, University of Alabama at Birmingham.</li>
</ul>

<h2 id="-current-position"><i class="fas fa-user-astronaut"></i> Current position</h2>

<p><strong>Instructor in Medicine</strong> - Dana-Farber Cancer Institute/Harvard Medical School.</p>

<p>My current work lies at the intersection of computational and systems biology. I am developing ways to better understand retroviral integration targeting biases, how those biases relate to provirus expression and disease progression, and how host factors influence said targeting biases. I pursue these questions using both wet and dry lab approaches. I am actively developing an improved computational and statistical framework for determining integration targeting biases within datasets and assessing differential targeting between them. I am additionally involved in designing, performing, and interpreting biophysical experiments related to understanding the role of liquid-liquid phase separation in HIV-1 infection and integration targeting. Beyond my particular research projects, I actively mentor undergraduate students, research technicians, and other trainees in both wet and dry lab research.</p>

<p><strong>Consulting</strong> - Freelance.</p>

<p>I have consulted companies on processing and wrangling large data files into smaller, more user-friendly formats. I am open to more consulting opportunities in data wrangling, bioinformatics, and data analysis. If interested, please <a href="mailto:gregoryjbedwell@gmail.com">get in touch</a>!</p>

<h2 id="-previous-positions"><i class="fas fa-map-marker-alt"></i> Previous positions</h2>

<p><strong>Post-doctoral fellow</strong> - Dana-Farber Cancer Institute/Harvard Medical School</p>

<p>As a post-doctoral fellow, I devised a recombinant protein expression system and purification scheme for the human protein CPSF6, a protein with low sequence complexity and strong compositional biases. This expression system has since been successfully applied to several similar protein targets. I additionally began my foray into conceptualizing and developing computational methods to better define HIV-1 integration targeting preferences. A strategy for categorizing genes into recurrently targeted and recurrently avoided populations was published in Nucleic Acids Research in 2021.</p>

<p><strong>Graduate student</strong> - University of Alabama at Birmingham.</p>

<p>As a graduate student, I was interested in understanding and exploiting the architecture of bacteriophage P22. I devised a system to selectively mineralize photocatalytic titanium dioxide within the pseudo-icosahedral shell of procapsid-like particles. Confining highly insoluble titanium dioxide within the capsid shell works to effectively ‚Äúsolubilize‚Äù the material, making it potentially more tractable for use in aqueous environments. In addition, I sought to understand the mechanism of so-called ‚Äúheadful‚Äù packaging. I was able to show that the P22 portal protein acts as a biological pressure sensor that responds to the increasing internal pressure introduced by increasing incorporation of dsDNA.</p>

<h2 id="-programming-and-statistics"><i class="fas fa-chart-bar"></i> Programming and statistics</h2>

<p><strong>Languages</strong>: Proficiency in R. Competency with Python, AWK, sed, etc. Exposure to Julia.</p>

<p><strong>Computing tools and technologies</strong>: bash scripting, git and GitHub, tidy data and R tidyverse, knitr/Rmarkdown/quarto/xaringan, LaTeX, HPC clusters (SLURM and Grid Engine), SageMath, jq.</p>

<p><strong>Bioinformatic tools</strong>:</p>
<ul>
  <li><u>Bioconductor</u>: GenomicRanges, Biostrings, BSgenome, SummarizedExperiment, Rsamtools, DESeq2, and many more.</li>
  <li><u>Miscellaneous bioinformatics software</u>: FastQC, bedtools, samtools, MACS2, and more.</li>
  <li><u>Sequence (pseudo)aligners</u>: Rsubread, BWA-MEM, STAR, kallisto.</li>
  <li><u>Multiple sequence alignment</u>: hh-suite, Clustal Omega, MUSCLE.</li>
  <li><u>Hydrodynamic modeling</u>: hullrad, HYDROPRO.</li>
  <li><u>Structure prediction</u>: MODELLER, AlphaFold, RoseTTAFold, ESMFold.</li>
  <li><u>Protein design</u>: RFdiffusion.</li>
</ul>

<p><strong>Statistics</strong>: discrete/categorical data analysis, linear and nonlinear least squares, generalized linear models, parametric and nonparametric models/tests, null hypothesis significance testing, bootstrapping, maximum likelihood estimation, optimization, hidden Markov models, etc.</p>

<h2 id="-software"><i class="fas fa-code"></i> Software</h2>

<ul>
  <li>
    <p><a href="https://github.com/gbedwell/calibrateR">calibrateR</a>: An R package written to streamline common laboratory calculations. A brief description of the package and general usage can be found <a href="https://gbedwell.github.io/calibrateR/">here</a>.</p>
  </li>
  <li>
    <p><a href="https://github.com/gbedwell/nbconv">nbconv</a>: An R package that implements three distinct methods for evaluating the sum of arbitrary negative binomial convolutions. See <a href="https://gbedwell.github.io/nb-convolutions/">this post</a> for more information. nbconv can be found on <a href="https://cran.r-project.org/web/packages/nbconv/index.html">CRAN</a>.</p>
  </li>
</ul>

<h2 id="-selected-publications"><i class="fas fa-align-left"></i> Selected Publications</h2>

<p>Bedwell, G.J. (2023). <strong>nbconv: Evaluate Arbitrary Negative Binomial Convolutions</strong>. <em>CRAN</em>.  <a href="http://doi.org/https://cran.r-project.org/web/packages/nbconv/index.html">https://cran.r-project.org/web/packages/nbconv/index.html</a>.  [üîì¬†Open Access.]</p>

<p>Mohammadi, A., Etemad, B., Zhang, X., Li, Y., Sharaf, R., Kittilson, A., Melberg, M., Melberg, C., Wong, C., Fajnzylber, J., Worrall, D.P., Rosenthal, A., Jordan, H., Jilg, N., Kaseke, C., Giguel, F., Lian, X., Deo, R., Gillespie, E., Chishti, R., Abrha, S., Adams, T., Siagian, A., Anderson, P.L., Deeks, S.G., Lederman, M.M., Yawetz, S., Kuritzkes, D.R., Lichterfeld, M.D., Tsibris, A., Carrington, M., Brumme,  Z.L., Castillo-Mancilla, J.R., Gaiha, G.D., &amp; Li, J.Z. (2023). <strong>Viral and Host Mediators of Non-Suppressible HIV-1 Viremia</strong>. <em>medRxiv</em>.  <a href="http://doi.org/10.1101/2023.03.30.23287124">10.1101/2023.03.30.23287124</a>.  [üîì¬†Open Access.]</p>

<p>Bedwell, G.J., Jang, S., Li, W., Singh, P.K., &amp; Engelman, A.N. (2021). <strong>rigrag: high-resolution mapping of genic targeting preferences during HIV-1 integration in vitro and in vivo</strong>. <em>Nucleic Acids Research</em>, <em>49</em>, 7330‚Äì7346.  <a href="http://doi.org/10.1093/nar/gkab514">10.1093/nar/gkab514</a>.  [üîì¬†Open Access.]</p>

<p>Bedwell, G.J., &amp; Engelman, A.N. (2021). <strong>Factors that mold the nuclear landscape of HIV-1 integration</strong>. <em>Nucleic Acids Research</em>, <em>49</em>, 621‚Äì635.  <a href="http://doi.org/10.1093/nar/gkaa1207">10.1093/nar/gkaa1207</a>.  [üîì¬†Open Access.]</p>

<p>Li, W., Singh, P.K., Sowd, G.A., Bedwell, G.J., Jang, S., Achuthan, V., Oleru, A.V., Wong, D., Fadel, H.J., Lee, K., KewalRamani, V.N., Poeschla, E.M., Herschhorn, A., &amp; Engelman, A.N. (2020). <strong>CPSF6-Dependent Targeting of Speckle-Associated Domains Distinguishes Primate from Nonprimate Lentiviral Integration</strong>. <em>mBio</em>, <em>11</em>, e02254-20.  <a href="http://doi.org/10.1128/mBio.02254-20">10.1128/mBio.02254-20</a>.  [üîì¬†Open Access.       ]</p>

<p>Zhang, D.-W., Luo, R.-H., Xu, L., Yang, L.-M., Xu, X.-S., Bedwell, G.J., Engelman, A.N., Zheng, Y.-T., &amp; Chang, S. (2019). <strong>A HTRF based competitive binding assay for screening specific inhibitors of HIV-1 capsid assembly targeting the C-Terminal domain of capsid</strong>. <em>Antiviral Research</em>, <em>169</em>, 104544.  <a href="http://doi.org/10.1016/j.antiviral.2019.104544">10.1016/j.antiviral.2019.104544</a>.</p>

<p>Jang, S., Cook, N.J., Pye, V.E., Bedwell, G.J., Dudek, A.M., Singh, P.K., Cherepanov, P., &amp; Engelman, A.N. (2019). <strong>Differential role for phosphorylation in alternative polyadenylation function versus nuclear import of SR-like protein CPSF6</strong>. <em>Nucleic Acids Research</em>, <em>47</em>, 4663‚Äì4683.  <a href="http://doi.org/10.1093/nar/gkz206">10.1093/nar/gkz206</a>.  [üîì¬†Open Access.]</p>

<p>Zhou, Z., Bedwell, G.J., Li, R., Palchoudhury, S., Prevelige, P.E. Jr., &amp; Gupta, A. (2017). <strong>Pathways for Gold Nucleation and Growth over Protein Cages</strong>. <em>Langmuir</em>, <em>33</em>, 5925‚Äì5931.  <a href="http://doi.org/10.1021/acs.langmuir.7b01298">10.1021/acs.langmuir.7b01298</a>.</p>

<p>Bedwell, G.J., &amp; Prevelige, P.E. Jr. (2017). <strong>Targeted mutagenesis of the P22 portal protein reveals the mechanism of signal transmission during DNA packaging</strong>. <em>Virology</em>, <em>505</em>, 127‚Äì138.  <a href="http://doi.org/10.1016/j.virol.2017.02.019">10.1016/j.virol.2017.02.019</a>.  [üîì¬†Open Access.]</p>

<p>Cherwa, J.E. Jr., Tyson, J., Bedwell, G.J., Brooke, D., Edwards, A.G., Dokland, T., Prevelige, P.E. Jr., &amp; Fane, B.F. (2017). <strong>œïX174 Procapsid Assembly: Effects of an Inhibitory External Scaffolding Protein and Resistant Coat Proteins In Vitro</strong>. <em>Journal of Virology</em>, <em>91</em>, e01878-16.  <a href="http://doi.org/10.1128/JVI.01878-16">10.1128/JVI.01878-16</a>.  [üîì¬†Open Access.]</p>

<p>Bedwell, G.J., Zhou, Z., Uchida, M., Dokland, T., Gupta, A., &amp; Prevelige, P.E. Jr. (2015). <strong>Selective biotemplated synthesis of TiO2 inside a protein cage</strong>. <em>Biomacromolecules</em>, <em>16</em>, 214‚Äì218.  <a href="http://doi.org/10.1021/bm501443e">10.1021/bm501443e</a>.</p>

<p>Bush, D.L., Monroe, E.B., Bedwell, G.J., Prevelige, P.E. Jr., Phillips, J.M., &amp; Vogt, V.M. (2014). <strong>Higher-order structure of the Rous sarcoma virus SP assembly domain</strong>. <em>Journal of Virology</em>, <em>88</em>, 5617‚Äì5629.  <a href="http://doi.org/10.1128/JVI.02659-13">10.1128/JVI.02659-13</a>.  [üîì¬†Open Access.    ]</p>

<p>Lucon, J., Qazi, S., Uchida, M., Bedwell, G.J., LaFrance, B., Prevelige, P.E. Jr., &amp; Dokland, T. (2012). <strong>Use of the interior cavity of the P22 capsid for site-specific initiation of atom-transfer radical polymerization with high-density cargo loading</strong>. <em>Nature Chemistry</em>, <em>4</em>, 781‚Äì788.  <a href="http://doi.org/10.1038/nchem.1442">10.1038/nchem.1442</a>.  [Cover story.]</p>

<p><br /></p>

<p><strong>A complete list of my peer-reviewed publications, can be found <a href="https://pubmed.ncbi.nlm.nih.gov/?term=%28Bedwell+GJ+AND+Prevelige+PE%29+OR+%28Bedwell+GJ+AND+Engelman%29+OR+%28Bedwell+GJ+AND+Schneider%29+OR+%28Bedwell+GJ+AND+Saad%29+OR+%28Bedwell+GJ+AND+Bedwell+DM%29">here</a>.</strong></p>


        
      </section>

      <footer class="page__meta">
        
        


        

      </footer>

      

      
    </div>

    
  </article>

  
  
</div>

    </div>

    

    <div id="footer" class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <ul class="social-icons">
    
      <li><strong>Follow:</strong></li>
    

    
      
        
      
        
      
        
      
        
      
        
      
        
      
    

    
      <li><a href="/feed.xml"><i class="fas fa-fw fa-rss-square" aria-hidden="true"></i> Feed</a></li>
    
  </ul>
</div>

<div class="page__footer-copyright">&copy; 2023 Greg Bedwell. Powered by <a href="https://jekyllrb.com" rel="nofollow">Jekyll</a> &amp; <a href="https://mademistakes.com/work/minimal-mistakes-jekyll-theme/" rel="nofollow">Minimal Mistakes</a>.</div>

      </footer>
    </div>

    
  <script src="/assets/js/main.min.js"></script>









<script type="text/javascript" async
	src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-MML-AM_CHTML">
</script>

<script type="text/javascript" async
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-MML-AM_CHTML">
  </script>

<script type="text/x-mathjax-config">
   MathJax.Hub.Config({
     extensions: ["tex2jax.js"],
     jax: ["input/TeX", "output/HTML-CSS"],
     tex2jax: {
       inlineMath: [ ['$','$'], ["\\(","\\)"] ],
       displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
       processEscapes: true
     },
     "HTML-CSS": { availableFonts: ["TeX"] }
   });
</script>



  </body>
</html>
